libs/agno/agno/workflow/v2/workflow.py

```diff
@@ -3368,6 +3387,19 @@ def _save_run_to_storage(self, workflow_run_response: WorkflowRunResponse) -> No
             self.workflow_session.upsert_run(workflow_run_response)
             self.write_to_storage()
 
+    def _store_member_run_if_enabled(self, step: Step, workflow_run_response: WorkflowRunResponse) -> None:
+        """Store agent/team responses in member_runs if enabled"""
+        if self.store_member_responses and step._executor_type in ["agent", "team"]:
+            # Get the raw response from the step's active executor
+            raw_response = getattr(step.active_executor, "run_response", None)
+            if raw_response and isinstance(raw_response, (RunResponse, TeamRunResponse)):
+                # Set parent_run_id to link to the workflow run
+                if hasattr(raw_response, "parent_run_id"):
+                    raw_response.parent_run_id = workflow_run_response.run_id
```

Comment:
So the agent/team run response should have a parent ID regardless?

Reply:
And it should be up to date by the time that run is stored, which happens inside the agent/team